### ISSUE TITLE
List navigation - Display flex and block level elements cause NVDA to not read the first list item contents in their entirety

### ISSUE THREAD

**Issue by patrickfox** [nvaccess/nvda#11491]
### Steps to reproduce:
Go to [this Codepen test page](https://codepen.io/pfox/pen/abNbvjz), which contains a variety of tests related to this issue.
Otherwise, create a list whose list items contain block-level elements and/or whose list items have a CSS display property value of "flex".
For example:
```
ul
  li.flex
    p First element in list item 1
    p Second element in list item 1
  li.flex
    p First element in list item 2
    p Second element in list item 2
  li.flex
    p First element in list item 3
    p Second element in list item 3
```

Navigate via lists using the "L" key

### Actual behavior:
Upon announcing the list and teh number of list items, NVDA announces only the first element in the first list item, skipping the remaining list item content which screen readers users may miss.

### Expected behavior:

NVDA should announce the presence of a list and the entire contents of the first list item.

### System configuration
Windows 10
NVDA 2020.2
Firefox 79.0
Chrome 84.0.4147.125

#### NVDA installed/portable/running from source:

Installed on Windows 10 VM, using Parallels on MacOS.

#### NVDA version:

2020.2

#### Windows version:

Windows 10 Enterprise Evaluation
Version 1809
OS Build 17763.1397

#### Name and version of other software in use when reproducing the issue:
Same as mentioned above

#### Other information about your system:

n/a

#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. 2020.1 exhibits the same issue.
#### If addons are disabled, is your problem still occuring?
I don't know, but I am not using an addons afaik
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by Adriani90** [comment]
cc: @feerrenrut, @MarcoZehe

**Comment by MarcoZehe** [comment]
CC @jcsteh

**Comment by jcsteh** [comment]
This is intentional. For elements which are generally of a readable size (e.g. links, list items), NVDA reads the entire element when using quick navigation. For container elements which can be very large such as lists and tables, NVDA reads only the first line. If there are two block elements in a list item, they'll be on separate lines, thus the behaviour you see.

**Comment by patrickfox** [comment]
Thanks for your response @jcsteh When navigating via list items, though, the entire list item contents are read, which seems inconsistent. The problem is that there no indication that there’s more content in the first list item that didn't get announced. 

Voiceover and JAWS both announce the presence of a list and the number of list items when using list navigation, but not the first list item. In these screen readers, users can then navigate the list items individually via list item navigation or element by element using the arrow keys. 

In NVDA, when using list navigation, each list is announced and the first list item is read by default - and only partially if the list item contains a block level element or has a flex display property applied to it. Once i encounter a list, I jump to list item navigation( the I key) but NVDA jumps to the second item and skips any remaining content in the first item that wasn't announced.

IMO, when navigating by lists, it would be an improvement to do one of the following:
a) not read or preview the first list item at all (similar to VO and JAWS)
b) read the entire first list item contents (but, the contents could be large...)
c) keep it as is, but when switching to list item navigation, repeat the first list item vs jumping to the second item

**Comment by XLTechie** [comment]
For what it's worth, I would vote for option A as being preferable to the 
current behavior, or option B if A is impractical.

The problem with option C, is that the user may not realize that the entire item 
was not spoken the first time. Then hearing the repeat of the start of the item, 
might skip right to the second item, without understanding that the first item 
was incomplete.
That is very dependent on the nature of the item, but it seems like a situation 
to be avoided.

**Comment by Adriani90** [comment]
This seems similar to what should happen on description lists aka ```<dl>``` with ```<dt>``` and ```<dd>```.
In general, list items are not that huge in practice, because for sighted people lists should also give a visual orientation and thus having huge list items would be anyway a bad web experience for sighted people as well.

Actually I would expect NVDA's behavior as follows:
1. Press l: NVDA reports "list with 5 items; complete list item"
2. When pressing i, NVDA should report the whole item as well.

I understand that this would be problematic in some situations as Jamie explained above, but those situations are rare and the same problems would be valid for sighted people as well.

The benefits of having the whole list item reported are still higher in my opinion at least.
